Library/Homebrew/os/mac.rb

```diff
@@ -94,6 +103,13 @@ def sdk(v = nil)
       @locator.sdk_if_applicable(v)
     end
 
+    def sdk_for_formula(f, v = nil)
+      # If the formula requires Xcode, don't return the CLT SDK
+      return Xcode.sdk if f&.requirements&.find { |req| req.is_a? XcodeRequirement }
```

Comment:
I should probably disallow `nil` here and do something like `formula ? MacOS.sdk_for_formula(formula) : MacOS.sdk` in the superenv.